modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/row/RowSchema.java

```diff
@@ -39,6 +43,43 @@ public List<TypeSpec> fields() {
         return fields;
     }
 
+    /**
+     * Gets an object representation from a tuple's field.
+     *
+     * @param fieldIndex Field index to read.
+     * @param tuple Tuple to read the value from.
+     * @return An object representation of the value.
+     */
+    public @Nullable Object value(int fieldIndex, InternalTuple tuple) {
```

Comment:
It would be better to swap arguments here. Almost every time field id follows the source data structure and not vice versa.

Reply:
Swapped, thanks.
p.s. I was guided by the order in which they go in RowHandler#get/set :roll_eyes: